### ISSUE TITLE
very frequent Timeout errors 

### ISSUE THREAD

**Issue by vivekanandbv** [seydx/homebridge-camera-ui#470]
receiving time out errors for prebuffer and video analysis every 5 minutes.
restarting homebridge sometimes resolves the issue, but happens again in a day or 2. 
although restarting doesn't work about half the times.
<img width="739" alt="Screen Shot 2022-02-24 at 9 10 18 PM" src="https://user-images.githubusercontent.com/79880739/155556877-4455c7a5-881d-4b0b-b0d6-1587a934cd09.png">

**Comment by MichlW** [comment]
Regarding the prebuffer timeouts I have exactly the same problems :-(

```
...
[2/24/2022, 6:05:30 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[2/24/2022, 6:05:30 PM] [CameraUI] Haustür: Prebufferring process closed
[2/24/2022, 6:05:30 PM] [CameraUI] Haustür: Restart prebuffer session..
[2/24/2022, 6:05:44 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 474 minutes
[2/24/2022, 6:06:39 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[2/24/2022, 6:06:39 PM] [CameraUI] Haustür: Prebufferring process closed
[2/24/2022, 6:06:39 PM] [CameraUI] Haustür: Restart prebuffer session..
[2/24/2022, 6:06:51 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 473 minutes
[2/24/2022, 6:07:48 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[2/24/2022, 6:07:48 PM] [CameraUI] Haustür: Prebufferring process closed
[2/24/2022, 6:07:48 PM] [CameraUI] Haustür: Restart prebuffer session..
[2/24/2022, 6:08:01 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 472 minutes
[2/24/2022, 6:08:58 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[2/24/2022, 6:08:58 PM] [CameraUI] Haustür: Prebufferring process closed
[2/24/2022, 6:08:58 PM] [CameraUI] Haustür: Restart prebuffer session..
[2/24/2022, 6:09:11 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 471 minutes
[2/24/2022, 6:10:08 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[2/24/2022, 6:10:08 PM] [CameraUI] Haustür: Prebufferring process closed
[2/24/2022, 6:10:08 PM] [CameraUI] Haustür: Restart prebuffer session..
[2/24/2022, 6:10:21 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 470 minutes
...
```

**Comment by platini76** [comment]
[01/03/2022, 02:30:54] [CameraUI] Camera Terrazzo: FFmpeg prebuffer process exited with error! (null) - [tee @ 0x3c28470] dimensions not set - Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument - Error initializing output stream 0:1 --  - 
[01/03/2022, 02:30:54] [CameraUI] Camera Terrazzo: Prebufferring process closed
[01/03/2022, 02:30:54] [CameraUI] Camera Terrazzo: Restart prebuffer session..
[01/03/2022, 02:30:56] [CameraUI] Camera Ballatoio: FFmpeg prebuffer process exited with error! (null) - [tee @ 0x25f7ae0] dimensions not set - Could not write header for output file #0 (incorrect codec parameters ?): Invalid argument - Error initializing output stream 0:1 --  -

**Comment by jamesbretz** [comment]
I was getting this with a Wyzecam V2 on RTSP firmware, rebooted the camera it and it seems to have sorted itself out (for the past few days at least).

**Comment by seydx** [comment]
its mostly due to wrong camera and/or config.json issues

can you guys post your config.json and/or screenshot from your camera settings pls

**Comment by MichlW** [comment]
My config.json:
```
{
            "name": "CameraUI",
            "port": 8081,
            "atHomeSwitch": false,
            "debug": false,
            "mqtt": {
                "active": true,
                "tls": false,
                "host": "172.17.0.1",
                "port": 1883
            },
            "http": {
                "active": false,
                "port": 7272,
                "localhttp": false
            },
            "smtp": {
                "active": false,
                "port": 2727,
                "space_replace": "+"
            },
            "ftp": {
                "active": false,
                "port": 5050,
                "useFile": false
            },
            "ssl": {
                "active": false
            },
            "options": {
                "videoProcessor": "ffmpeg"
            },
            "cameras": [
                {
                    "name": "Haustür",
                    "manufacturer": "YI Technology",
                    "model": "Outdoor 1080p",
                    "serialNumber": "xxxxxxxxx",
                    "motion": true,
                    "doorbell": true,
                    "switches": true,
                    "motionTimeout": 0,
                    "unbridge": true,
                    "hsv": true,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "subSource": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "stillImageSource": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "rtspTransport": "udp",
                        "maxStreams": 2,
                        "maxWidth": 0,
                        "maxHeight": 0,
                        "maxFPS": 0,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "encoderOptions": "-preset ultrafast -tune zerolatency",
                        "audio": false
                    },
                    "mqtt": {
                        "motionTopic": "cameras/motion",
                        "motionMessage": "Haustuer_MotionStart",
                        "motionResetTopic": "cameras/motion",
                        "motionResetMessage": "Haustuer_MotionStop",
                        "doorbellTopic": "doorbell",
                        "doorbellMessage": "Haustuer_Doorbell"
                    },
                    "smtp": {
                        "email": "Haustür"
                    },
                    "videoanalysis": {
                        "active": false
                    }
                }
            ],
            "platform": "CameraUI"
        }
```

And the setting from yi-hack-v5:
![image](https://user-images.githubusercontent.com/49558935/162675785-6a4ff13f-4852-4bad-9187-b6e1b68b0d97.png)
![image](https://user-images.githubusercontent.com/49558935/162675732-37a2e9fb-fc3c-40a3-941c-6bfcf173b3d8.png)
![image](https://user-images.githubusercontent.com/49558935/162677082-356e1b48-bee4-47e3-bf3a-fb8d50ebb6d5.png)

**Comment by seydx** [comment]
@MichlW 

try 

```json
{
            "name": "CameraUI",
            "port": 8081,
            "atHomeSwitch": false,
            "debug": false,
            "mqtt": {
                "active": true,
                "tls": false,
                "host": "172.17.0.1",
                "port": 1883
            },
            "http": {
                "active": false,
                "port": 7272,
                "localhttp": false
            },
            "smtp": {
                "active": false,
                "port": 2727,
                "space_replace": "+"
            },
            "ftp": {
                "active": false,
                "port": 5050,
                "useFile": false
            },
            "ssl": {
                "active": false
            },
            "options": {
                "videoProcessor": "ffmpeg"
            },
            "cameras": [
                {
                    "name": "Haustür",
                    "manufacturer": "YI Technology",
                    "model": "Outdoor 1080p",
                    "serialNumber": "xxxxxxxxx",
                    "motion": true,
                    "doorbell": true,
                    "switches": true,
                    "motionTimeout": 0,
                    "unbridge": true,
                    "hsv": true,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "subSource": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "stillImageSource": "-i rtsp://172.16.2.50/ch0_0.h264",
                        "rtspTransport": "tcp",
                        "maxStreams": 2,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "audio": false
                    },
                    "mqtt": {
                        "motionTopic": "cameras/motion",
                        "motionMessage": "Haustuer_MotionStart",
                        "motionResetTopic": "cameras/motion",
                        "motionResetMessage": "Haustuer_MotionStop",
                        "doorbellTopic": "doorbell",
                        "doorbellMessage": "Haustuer_Doorbell"
                    },
                    "smtp": {
                        "email": "Haustür"
                    },
                    "videoanalysis": {
                        "active": false
                    }
                }
            ],
            "platform": "CameraUI"
        }
```

**Comment by MichlW** [comment]
I tried your suggestions in the videoConfig part and got prebuffer errors again:

```
[4/11/2022, 12:30:08 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 810 minutes
[4/11/2022, 12:30:08 PM] [CameraUI] camera.ui v5.0.16 is listening on port 8081 (http)
[4/11/2022, 12:30:53 PM] [CameraUI] Haustür: Starting video stream: native
[4/11/2022, 12:30:59 PM] [CameraUI] Haustür: Stopped video stream.
[4/11/2022, 12:33:47 PM] [CameraUI] Haustür: Starting video stream: native
[4/11/2022, 12:33:51 PM] [CameraUI] Haustür: Stopped video stream.
[4/11/2022, 12:33:52 PM] [CameraUI] Haustür: Starting video stream: native
[4/11/2022, 12:33:54 PM] [CameraUI] Haustür: Stopped video stream.
[4/11/2022, 12:35:33 PM] [Homebridge UI] [homebridge-camera-ui] Incoming Request: /interfaceConfig
[4/11/2022, 12:36:48 PM] [CameraUI] Haustür: Starting video stream: native
[4/11/2022, 12:37:15 PM] [Homebridge UI] [homebridge-camera-ui] Terminating child process...
[4/11/2022, 12:37:15 PM] [Homebridge UI] [homebridge-camera-ui] Child process ended
[4/11/2022, 12:37:31 PM] [CameraUI] Haustür: Stopped video stream.
[4/11/2022, 12:55:59 PM] [CameraUI] Haustür: Prebuffer timed out... killing ffmpeg session
[4/11/2022, 12:55:59 PM] [CameraUI] Haustür: Prebufferring process closed
[4/11/2022, 12:55:59 PM] [CameraUI] Haustür: Restart prebuffer session..
[4/11/2022, 12:56:26 PM] [CameraUI] Haustür: Prebuffering scheduled for restart at 2AM: 784 minutes
```

In the past few days I also tested the robustness of the rtsp stream and could receive several hours of stream in VLC and in my Kodi plugin on TV without problems.
Therefore I think that the rtsp source is not the problem.

**Comment by vivekanandbv** [comment]
{
            "name": "CameraUI",
            "port": 8081,
            "atHomeSwitch": false,
            "debug": false,
            "mqtt": {
                "active": false,
                "tls": false,
                "port": 1883
            },
            "http": {
                "active": false,
                "port": 7272,
                "localhttp": false
            },
            "smtp": {
                "active": false,
                "port": 2727,
                "space_replace": "+"
            },
            "ftp": {
                "active": false,
                "useFile": false,
                "port": 5050
            },
            "ssl": {
                "active": false
            },
            "options": {
                "videoProcessor": "/usr/local/lib/node_modules/homebridge-camera-ui/node_modules/ffmpeg-for-homebridge/ffmpeg"
            },
            "cameras": [
                {
                    "name": "Parking",
                    "motion": false,
                    "switches": true,
                    "motionTimeout": 15,
                    "unbridge": true,
                    "hsv": false,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://admin:admin@192.168.1.3:554/?chID=1&streamType=sub&linkType=udp",
                        "subSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=1&streamType=sub&linkType=udp",
                        "stillImageSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=1&streamType=sub&linkType=udp",
                        "rtspTransport": "udp",
                        "maxStreams": 2,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "audio": false,
                        "debug": false
                    },
                    "smtp": {
                        "email": "Parking"
                    },
                    "videoanalysis": {
                        "active": false
                    },
                    "mqtt": {}
                },
                {
                    "name": "Main Door",
                    "motion": true,
                    "doorbell": true,
                    "switches": true,
                    "useInterfaceTimer": false,
                    "motionTimeout": 0,
                    "unbridge": true,
                    "hsv": true,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://admin:admin@192.168.1.3:554/?chID=3&streamType=sub&linkType=udp",
                        "subSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=3&streamType=sub&linkType=udp",
                        "stillImageSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=3&streamType=sub&linkType=udp",
                        "maxStreams": 2,
                        "maxFPS": 10,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "audio": false,
                        "debug": true,
                        "debugReturn": false
                    },
                    "smtp": {
                        "email": "Main Door"
                    },
                    "videoanalysis": {
                        "active": true
                    },
                    "mqtt": {}
                },
                {
                    "name": "Street",
                    "motion": true,
                    "useInterfaceTimer": false,
                    "motionTimeout": 0,
                    "unbridge": true,
                    "hsv": true,
                    "prebuffering": true,
                    "prebufferLength": 4,
                    "videoConfig": {
                        "source": "-i rtsp://admin:admin@192.168.1.3:554/?chID=2&streamType=sub&linkType=udp",
                        "subSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=2&streamType=sub&linkType=udp",
                        "stillImageSource": "-i rtsp://admin:admin@192.168.1.3:554/?chID=2&streamType=sub&linkType=udp",
                        "maxStreams": 2,
                        "maxFPS": 10,
                        "vcodec": "copy",
                        "acodec": "libfdk_aac",
                        "audio": false,
                        "debug": true
                    },
                    "smtp": {
                        "email": "Street"
                    },
                    "videoanalysis": {
                        "active": true
                    },
                    "mqtt": {}
                }
            ],
            "platform": "CameraUI"
        }
<img width="717" alt="Screen Shot 2022-04-13 at 13 16 30" src="https://user-images.githubusercontent.com/79880739/163126379-d39dc55a-4fe1-49f8-a5d4-0dc8d61f0195.png">
,

**Comment by seydx** [comment]
Timeout errors occur due to your network or connection to the camera

With the current beta v5.0.17-beta.4 I have made some improvements here

**Comment by Cylus7** [comment]
Hi, I have this same issue-  @vivekanandbv did it come right for you with that update?

**Comment by vivekanandbv** [comment]
> Hi, I have this same issue- @vivekanandbv did it come right for you with that update?

Yes, I don’t have this issue anymore.

**Comment by vivekanandbv** [comment]
I recently rerinstalled home bridge and had to reconfigure everything. What I noticed was that everything worked fine when the Mac was awake, but timeout errors started occurring about 5 mins after the display goes off. I figured it was because the homebridge machine was going to sleep. If on Mac, enabling 'prevent automatic sleeping when display is off' option under energy tab of advanced display settings fixes the issue. I believe if homebridge running devices are kept from going to sleep, these issues will be resolved.
